lighthouse-core/test/audits/byte-efficiency/render-blocking-resources-test.js

```diff
@@ -87,5 +87,41 @@ describe('Render blocking resources audit', () => {
       // Saving 1000 + 1000 + 100ms for TCP handshake + 1 RT savings + server response time
       assert.equal(result, 2100);
     });
+
+    it('computes savings for stylesheets deferred by AMP', () => {
+      const recordSlow = props => {
+        const parsedURL = {host: 'slow.com', securityOrigin: 'http://slow.com'};
+        return Object.assign({parsedURL, requestId: requestId++}, props);
+      };
+      const serverResponseTimeByOrigin = new Map([
+        ['http://example.com', 100],
+        ['http://slow.com', 4000],
+      ]);
+      const Stacks = [{
+        detector: 'js',
+        id: 'amp',
+        name: 'AMP',
+        version: '2006180239003',
+        npm: 'https://www.npmjs.com/org/ampproject',
+      }];
+      const simulator = new Simulator({rtt: 1000, serverResponseTimeByOrigin});
+      const documentNode = new NetworkNode(record({transferSize: 4000}));
+      const styleNode = new NetworkNode(recordSlow({
+        transferSize: 3000,
+        resourceType: NetworkRequest.TYPES.Stylesheet,
+      }));
+      const styleNode2 = new NetworkNode(recordSlow({
+        transferSize: 3000,
+        resourceType: NetworkRequest.TYPES.Stylesheet,
+      }));
+      const deferredIds = new Set([2, 3]);
+      const wastedBytesMap = new Map();
+
+      documentNode.addDependent(styleNode);
+      documentNode.addDependent(styleNode2);
+      const result = estimate(simulator, documentNode, deferredIds, wastedBytesMap, Stacks);
+      // Savings capped at 2100 for AMP stylesheets since they are run async after timeout
+      assert.equal(result, 2100);
```

Comment:
for this test if my math is right I would expect the savings cap to be more like ~0s on AMP

you might want to add a second test with a lower overall RTT passed to the simulator because right now the main document won't finish until ~2.1s (1 RTT for TCP handshake, 100ms for server response time, 1 RTT for request)

Reply:
You are right that the savings is ~0s however I see the main document finishing at ~4.1s (4 RTT and 100ms server response).  What are the other 2 RTT ~for~?

Reply:
hm I'm not sure why that is off top of my head, we can dig into it together

Reply:
oh I forgot we added DNS time by default :) 2 more RTTs for DNS